lib/portlayer/exec/container.go

```diff
@@ -326,10 +337,8 @@ func (c *Container) start(ctx context.Context) error {
 	if c.vm == nil {
 		return fmt.Errorf("vm not set")
 	}
-	// get existing state and set to starting
-	// if there's a failure we'll revert to existing
-	finalState := c.updateState(StateStarting)
-	defer func() { c.updateState(finalState) }()
+	// Set state to Starting
+	c.updateState(StateStarting)
```

Comment:
Yes, it's been racy before this change as well. However, I'm not sure if we can decide on one initial state here - on a `run -it busybox`, the initial state here is `Created`. It could be `Stopped` if the container is stopped.